synapse/http/matrixfederationclient.py

```diff
@@ -174,6 +174,16 @@ async def _handle_json_response(
         d = timeout_deferred(d, timeout=timeout_sec, reactor=reactor)
 
         body = await make_deferred_yieldable(d)
+    except ValueError as e:
+        # The JSON content was invalid.
```

Comment:
should we not catch `JSONDecodeError` here?

Reply:
Arg, I forget half the time there's a specific error for this. Thank you!

Reply:
Ah, so we need to catch `ValueError` since we manually throw that when doing JSON validation:

https://github.com/matrix-org/synapse/blob/b6ca69e4f109c745f022885ecb8aa86255f84ecf/synapse/util/__init__.py#L30-L32

I had looked at raising a `JSONDecodeError` there, but you don't really have the necessary information, since we don't have the position of what is being parsed.

A couple of options:

* Leave this as a `ValueError`
* Change the error raised by `_reject_invalid_json` to a custom exception `InvalidJsonError` (which inherits from `ValueError`) and catch both here.
* Raise `JSONDecodeError` in `_reject_invalid_json` by giving a dummy position.

What do you think?

Reply:
I think I don't feel very strongly. May as well leave it as-is.